master/internal/api_trials.go

```diff
@@ -501,13 +501,16 @@ func (a *apiServer) GetTrial(_ context.Context, req *apiv1.GetTrialRequest) (
 	switch err := a.m.db.QueryProtof(
 		"proto_get_trial_workloads",
 		[]interface{}{
+			"total_batches",
+			db.OrderByToSQL(apiv1.OrderBy_ORDER_BY_ASC),
 			db.OrderByToSQL(apiv1.OrderBy_ORDER_BY_ASC),
 			db.OrderByToSQL(apiv1.OrderBy_ORDER_BY_ASC),
 		},
 		&workloads,
 		req.TrialId,
-		nil,
-		nil,
+		0,
+		1000000,
+		"All",
```

Comment:
so we are assuming trial has <= million batches? What, if anything, goes wrong if this assumption is not correct?

maybe it's not so important because if you care about workloads you should be using the workloads endpoint? 

it looks like we are still getting workloads as a part of trial API. is the plan to get rid of it eventually?